Upstream ImageMagick changed part of the API and did not updated their
deprecated support stuff, so bump us along for now to avoid having to downgrade.

Gentoo-bug: 247292
Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar --exclude '*~' xine-lib-1.1.15.orig/src/libxinevdec/image.c xine-lib-1.1.15/src/libxinevdec/image.c
--- xine-lib-1.1.15.orig/src/libxinevdec/image.c	2008-06-14 16:15:00.000000000 -0700
+++ xine-lib-1.1.15/src/libxinevdec/image.c	2008-11-26 18:10:16.416309036 -0800
@@ -110,7 +110,15 @@
     width = MagickGetImageWidth(wand) & ~1; /* must be even for init_yuv_planes */
     height = MagickGetImageHeight(wand);
     img_buf = malloc(width * height * 3);
+/* In 6.4.5.4 MagickGetImagePixels changed to MagickGetAuthenticPixels 
+ * But upstream did not update their deprecated compat stuff.
+ * So do a fun hack to make it work.
+ * - 2008/11/26 Robin H. Johnson <robbat2@gentoo.org>
+ */
+#if MagickLibVersion >= 0x645
+#define MagickGetImagePixels MagickGetAuthenticPixels
+#endif
     MagickGetImagePixels(wand, 0, 0, width, height, "RGB", CharPixel, img_buf);
     DestroyMagickWand(wand);
 
     _x_stream_info_set(this->stream, XINE_STREAM_INFO_VIDEO_WIDTH, width);
